python/cudf/cudf/tests/test_json.py

```diff
@@ -1177,20 +1177,15 @@ def test_chunked_nested_json_reader(self, tag, data, chunk_size):
         df = cudf.concat(chunks, ignore_index=True)
         assert expected.to_arrow().equals(df.to_arrow())
 
-    def test_order_nested_json_reader(self, tag, data):
+    def test_order_nested_json_reader(self, request, tag, data):
         expected = pd.read_json(StringIO(data), lines=True)
         target = cudf.read_json(StringIO(data), lines=True)
-        if tag == "dtype_mismatch":
-            with pytest.raises(AssertionError):
-                # pandas parses integer values in float representation
-                # as integer
-                assert pa.Table.from_pandas(expected).equals(target.to_arrow())
-        elif tag == "missing":
-            with pytest.raises(AssertionError):
-                # pandas inferences integer with nulls as float64
-                assert pa.Table.from_pandas(expected).equals(target.to_arrow())
-        else:
-            assert pa.Table.from_pandas(expected).equals(target.to_arrow())
+        request.applymarker(
+            pytest.mark.xfail(
+                tag == "dtype_mismatch", reason="int vs float mismatch"
+            )
+        )
+        assert_eq(expected, target)
```

Comment:
Can pandas now handle comparing the equality of nested types/values? I remember the last time I checked pandas wasn't able to that properly and thus we resorted to use pyarrow for comparisons in case of nested types.

Reply:
I just was able to triage the issue here, looks like `pd.read_json` has a regression where the dataframe is expected to have a `RangeIndex` but a materialized `Index` of `int64` dtype is being returned in `pandas-2.2`. We should just change this test to the following:
```diff
diff --git a/python/cudf/cudf/tests/test_json.py b/python/cudf/cudf/tests/test_json.py
index ec980adc33..5a459e98d1 100644
--- a/python/cudf/cudf/tests/test_json.py
+++ b/python/cudf/cudf/tests/test_json.py
@@ -1179,6 +1179,9 @@ class TestNestedJsonReaderCommon:
 
     def test_order_nested_json_reader(self, tag, data):
         expected = pd.read_json(StringIO(data), lines=True)
+        if PANDAS_GE_200:
+            # TODO: Remove after bug fix: <Pandas-bug-URL>
+            expected = expected.reset_index(drop=True)
         target = cudf.read_json(StringIO(data), lines=True)
         if tag == "dtype_mismatch":
             with pytest.raises(AssertionError):
```

Reply:
Ah I see. Yeah testing of nested data isn't well tested in pandas so probably better to use pyarrow to compare here. Will incorporate your change